weatherApp/App/LocationService/LocationService.swift

```diff
@@ -7,31 +7,29 @@
 //
 
 import CoreLocation
+import RxSwift
+import RxCoreLocation
 
-class LocationService: NSObject, LocationServiceProtocol {
-    
-    private var locationManager: CLLocationManager!
-    
-    var completion: LocationServiceCompletion!
+class LocationService: LocationServiceProtocol {
+
+    private (set) var location: Observable<Coordinates>
+
+    private let locationManager = CLLocationManager()
     
-    func getLocation(completion: @escaping LocationServiceCompletion) {
-        self.completion = completion
+    init() {
+        locationManager.distanceFilter = kCLDistanceFilterNone;
+        locationManager.desiredAccuracy = kCLLocationAccuracyBestForNavigation;
+        
+        location = locationManager.rx
+            .didUpdateLocations
+            .filter { $1.count > 0 }
+            .map {
+                let coord = $1.last!.coordinate
```

Comment:
In some cases is ok to use $ closure arguments like in the filter above. But for readability please use parameter names here.